packages/x-components/src/components/sliding-panel.vue

```diff
@@ -39,70 +39,111 @@
   import { Debounce } from './decorators/debounce.decorators';
 
   /**
-   * Horizontal slide panel component. The content that is provided to this component would be
-   * navigable horizontally using two navigational buttons.
+   * This component allows for any other component or element inside it to be horizontally
+   * navigable. It also implements customizable buttons as well as other minor customizations to its
+   * general behavior.
    *
    * @public
    */
   @Component
+  /* eslint-disable @typescript-eslint/unbound-method */
   export default class SlidingPanel extends Vue {
     /**
      * Scroll factor that will dictate how much the scroll moves when pressing a navigation button.
+     *
+     * @public
      */
     @Prop({ default: 0.7 })
-    protected scrollFactor!: number;
+    public scrollFactor!: number;
 
     /**
      * Would make the navigation buttons visible when they're needed or always hide them.
+     *
+     * @public
      */
     @Prop({ default: true })
-    protected showButtons!: boolean;
+    public showButtons!: boolean;
 
     /**
-     * MutationObserver that watches for changes inside the wrapping div to update
-     * the scroll position.
+     *
+     * @public
      */
-    protected scrollObserver = new MutationObserver(
-      // eslint-disable-next-line @typescript-eslint/unbound-method
-      this.restoreAndUpdateScroll
-    );
+    @Prop({ default: true })
+    public resetOnContentChange!: boolean;
 
     /**
      * Indicates if the scroll is at the start of the sliding panel.
+     *
+     * @internal
      */
     protected isScrollAtStart = true;
 
     /**
      * Indicates if the scroll is at the end of the sliding panel.
+     *
+     * @internal
      */
     protected isScrollAtEnd = true;
 
     /**
      * HTMLElement referencing the scroll of the component.
+     *
+     * @internal
      */
     public $refs!: {
       scrollContainer: HTMLElement;
     };
 
+    /**
+     * CSS classes to apply based on the scroll position.
+     *
+     * @returns The CSS classes to apply.
+     *
+     * @internal
+     */
+    protected get cssClasses(): VueCSSClasses {
+      return {
+        'x-sliding-panel--at-start': this.isScrollAtStart,
+        'x-sliding-panel--at-end': this.isScrollAtEnd
+      };
+    }
+
+    /**
+     * Initialises browser platform code:
+     * - Creates a mutation observer to detect content changes and reset scroll position.
+     * - Stores initial size and scroll position values.
+     *
+     * @internal
+     */
     mounted(): void {
-      this.scrollObserver.observe(this.$refs.scrollContainer, {
-        attributes: false,
-        childList: true,
-        subtree: true,
-        characterData: false
+      const resizeObserver = new ResizeObserver(this.debouncedUpdateScrollPosition);
+      resizeObserver.observe(this.$el);
+
+      const contentChangedObserver = new MutationObserver(this.restoreAndUpdateScroll);
```

Comment:
Browser APIs like `ResizeObserver` or `MutationObserver` shouldn't be used before the `mounted` hook.